server/src/com/cloud/network/IpAddressManagerImpl.java

```diff
@@ -1714,8 +1714,13 @@ public boolean applyStaticNats(List<? extends StaticNat> staticNats, boolean con
             return true;
         }
 
-        // get the list of public ip's owned by the network
-        List<IPAddressVO> userIps = _ipAddressDao.listByAssociatedNetwork(network.getId(), null);
+        // get the list of public ip's that need to be applied for the static NAT. manipulating only these
+        // ips prevents concurrency issues when disabling static nat at the same time.
+        List<IPAddressVO> userIps = new ArrayList<>();
```

Comment:
Hi @ProjectMoon, what about creating a method that returns the userIps list. This new method would contain lines 1719-1722. Then, you would be able to write a unit test and add lines 1717-1718 as its java documentation.